tests/general/configproviders/env_test.go

```diff
@@ -41,14 +45,38 @@ service:
       receivers: [hostmetrics]
       exporters: [otlp]
 `
-	testutils.AssertAllMetricsReceived(
-		t, "memory.yaml", "", nil,
-		[]testutils.CollectorBuilder{
-			func(collector testutils.Collector) testutils.Collector {
-				return collector.WithEnv(
-					map[string]string{"SOME_ENV_VAR": config},
-				).WithArgs("--config", "env:SOME_ENV_VAR")
-			},
-		},
-	)
+	tc := testutils.NewTestcase(t)
+	defer tc.PrintLogsOnFailure()
+	defer tc.ShutdownOTLPReceiverSink()
+
+	_, shutdown := tc.SplunkOtelCollector("", func(collector testutils.Collector) testutils.Collector {
+		return collector.WithEnv(
+			map[string]string{"SOME_ENV_VAR": config},
+		).WithArgs("--config", "env:SOME_ENV_VAR")
+	})
+	defer shutdown()
+
+	missingMetrics := map[string]struct{}{
+		"system.memory.usage": {},
+	}
+
+	assert.EventuallyWithT(t, func(tt *assert.CollectT) {
+		for i := 0; i < len(tc.OTLPReceiverSink.AllMetrics()); i++ {
+			m := tc.OTLPReceiverSink.AllMetrics()[i]
+			for j := 0; j < m.ResourceMetrics().Len(); j++ {
+				rm := m.ResourceMetrics().At(j)
+				for k := 0; k < rm.ScopeMetrics().Len(); k++ {
+					sm := rm.ScopeMetrics().At(k)
+					for l := 0; l < sm.Metrics().Len(); l++ {
+						delete(missingMetrics, sm.Metrics().At(l).Name())
```

Comment:
I am a little confused. Can you clarify how deleting/checking for missingMetric is helpful for this test case?

Reply:
that's what resourceMetrics was doing ; over the life of the ingest, check which metrics are returned and make sure they match.

Reply:
it's a straight replacement - it just happens we have one metric to check on, but the same structure is used for other tests. We can refactor this out later to make it a call to a testutils method if we like.

Reply:
Ah, I thought with the memory metric specifically named missing metric we were doing something different.